values.yaml

```diff
@@ -2,26 +2,33 @@
 # This is a YAML-formatted file.
 # Declare variables to be passed into your templates.
 
-image: quay.io/infinispan/server:12.1.3
+image: quay.io/infinispan/server:12.1.3.Final
 initContainerImage: registry.access.redhat.com/ubi8-micro
+testContainerImage: quay.io/repository/quay/busybox
 
 replicas: 1
 
-service:
-  type: DataGrid
-  container:
-    storage: 1Gi
+container:
+  extraJvmOpts: ""
+  storage:
+    size: 1Gi
     storageClassName: ""
-    ephemeralStorage: false
+    ephemeral: false
   resources:
-    extraJvmOpts: ""
-    cpu: 500m
-    memory: 512Mi
+    limits:
+      cpu: 500m
+      memory: 512Mi
+    requests:
+      cpu: 500m
+      memory: 512Mi
 
 security:
-  endpointAuthentication: true
-  endpointSecretName: ""
-  endpointSecretPassword: ""
+  authentication: true
+  secretName: ""
+  identities:
```

Comment:
With this is the following possible `helm install --set-file security.identitites=some/file`?

Reply:
This should work, but I can't to get it work, the flexibility of the Helm is not the same as we can do on the operator level.
Very strange error, seems to Helm see `credentials` as map, not array. 
```
coalesce.go:196: warning: cannot overwrite table with non table for identities (map[credentials:[map[password: username:]]])
``` 
Tried with different options.

Reply:
```
helm install example-infinispan . -n helm --set security.identities.credentials[0].username="developer" --set security.identities.credentials[0].password="developer"
```
working fine

Reply:
It's possible to get around these limitations by not processing the `security.identities` value as yaml.

If you update values.yaml to:

```yaml
security:
  authentication: true
  secretName: ""
  identities: ""
```

and secret.yaml to:

```yaml
{{- if not .Values.security.secretName }}
  {{- if .Release.IsInstall }}
{{- $password := randAlphaNum 8 }}
apiVersion: v1
kind: Secret
metadata:
  name: {{ printf "%s-generated-secret" (include "infinispan-helm-charts.name" .) }}
  labels:
    app: infinispan-secret-identities
  {{- include "infinispan-helm-charts.labels" . | nindent 4 }}
  annotations:
    "helm.sh/resource-policy": no-update
type: Opaque
{{- if .Values.security.identities }}
data:
  identities.yaml: |-
    {{ .Values.security.identities | b64enc }}
{{- else }}
stringData:
  identities.yaml: |
  username: developer
  password: {{ $password }}
  identities.yaml: |-
    credentials:
    - username: developer
      password: {{ $password }}
      roles:
        - admin
  {{- end }}
  {{- end }}
  {{- end }}
```

Reply:
Thanks, @ryanemerson That's good idea and solution.